java/src/main/java/org/rocksdb/ParsedEntryInfo.java

```diff
@@ -0,0 +1,88 @@
+package org.rocksdb;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Class to parse internal key to extract user key, entry type, sequence number.
+ */
+public class ParsedEntryInfo extends RocksObject {
+  protected ParsedEntryInfo() {
+    super(newParseEntryInstance());
+  }
+
+  @Override
+  protected void disposeInternal(final long handle) {
+    disposeInternalJni(handle);
+  }
+
+  public EntryType getEntryType() {
+    assert (isOwningHandle());
+    return EntryType.getEntryType(getEntryTypeJni(nativeHandle_));
+  }
+
+  public long getSequenceNumber() {
+    return getSequenceNumberJni(nativeHandle_);
+  }
+
+  public byte[] getUserKey() {
+    assert (isOwningHandle());
+    return userKeyJni(nativeHandle_);
+  }
+
+  public int userKey(final ByteBuffer key) {
+    if (key == null) {
+      throw new IllegalArgumentException("ByteBuffer parameters must not be null");
+    }
+    assert (isOwningHandle());
+    final int result;
+    if (key.isDirect()) {
+      result = userKeyDirect(nativeHandle_, key, key.position(), key.remaining());
+    } else {
+      result = userKeyByteArray(
+          nativeHandle_, key.array(), key.arrayOffset() + key.position(), key.remaining());
+    }
+    key.limit(Math.min(key.position() + result, key.limit()));
+    return result;
+  }
+
+  public void parseEntry(Options options, byte[] internalKey) {
```

Comment:
Please mark parameters as `final`. (and methods/functions below)

Reply:
done